rib/rib.go

```diff
@@ -97,6 +110,11 @@ type RIB struct {
 	// same AddXXX methods can be used along with network instance the operation
 	// referred to. The map is keyed by the operation ID.
 	pendingEntries map[uint64]*pendingEntry
+
+	// resolvedEntryHook is a function that is called for all entries that
+	// can be fully resolved in the RIB. In the current implementation it
+	// is called only for IPv4 entries.
+	resolvedEntryHook ResolvedEntryFn
```

Comment:
nit: is it conceptually only called for v4 or we are just doing it for v4 now? If latter then maybe we can call this v4ResolvedEntryHook?

Reply:
We are just doing it for v4 right now, it should be used for IPv6, MPLS and PBR going forward - hence keeping the name a bit more general.